mayastor-test/e2e/uninstall/uninstall_test.go

```diff
@@ -0,0 +1,193 @@
+					}
+				}
+			}
+		}
+	}
+
+	/// TODO Refine how we workout the address of the test-registry
+
+	/// If there is master node, use its IP address as the registry IP address
+	if len(master) != 0 {
+		return master, nil
+	}
+
+	if len(nodeIPs) == 0 {
+		return "", errors.New("no usable nodes found")
+	}
+
+	/// Choose the IP address of first node in the list as the registry IP address
+	return nodeIPs[0], nil
+}
+
+// Encapsulate the logic to find where the deploy yamls are
+func getDeployYamlDir() string {
+	_, filename, _, _ := runtime.Caller(0)
+	return path.Clean(filename + "/../../../../deploy")
+}
+
+// Helper for passing yaml from the deploy directory to kubectl
+func deleteDeployYaml(filename string) {
+	cmd := exec.Command("kubectl", "delete", "-f", filename)
+	cmd.Dir = getDeployYamlDir()
+	_, err := cmd.CombinedOutput()
+	Expect(err).ToNot(HaveOccurred())
+}
+
+// Encapsulate the logic to find where the templated yamls are
+func getTemplateYamlDir() string {
+	_, filename, _, _ := runtime.Caller(0)
+	return path.Clean(filename + "/../../install/deploy")
+}
+
+func makeImageName(registryAddress string, registryport string, imagename string, imageversion string) string {
+	return registryAddress + ":" + registryport + "/mayadata/" + imagename + ":" + imageversion
+}
+
+func deleteTemplatedYaml(filename string, imagename string, registryAddress string) {
+	fullimagename := makeImageName(registryAddress, "30291", imagename, "ci")
+	bashcmd := "IMAGE_NAME=" + fullimagename + " envsubst < " + filename + " | kubectl delete -f -"
+	cmd := exec.Command("bash", "-c", bashcmd)
+	cmd.Dir = getTemplateYamlDir()
+	_, err := cmd.CombinedOutput()
+	Expect(err).ToNot(HaveOccurred())
+}
+
+// We expect this to fail a few times before it succeeds,
+// so no throwing errors from here.
+func mayastorReadyPodCount() int {
+	var mayastorDaemonSet appsv1.DaemonSet
+	if k8sClient.Get(context.TODO(), types.NamespacedName{Name: "mayastor", Namespace: "mayastor"}, &mayastorDaemonSet) != nil {
+		return -1
+	}
+	return int(mayastorDaemonSet.Status.CurrentNumberScheduled)
+}
+
+// Teardown mayastor on the cluster under test.
+// We deliberately call out to kubectl, rather than constructing the client-go
+// objects, so that we can verfiy the local deploy yamls are correct.
+func teardownMayastor() {
+	registryAddress, err := getRegistryAddress()
+	Expect(err).ToNot(HaveOccurred())
+	deleteTemplatedYaml("mayastor-daemonset.yaml.template", "mayastor", registryAddress)
+	deleteTemplatedYaml("moac-deployment.yaml.template", "moac", registryAddress)
+	deleteTemplatedYaml("csi-daemonset.yaml.template", "mayastor-csi", registryAddress)
+	deleteDeployYaml("nats-deployment.yaml")
+	deleteDeployYaml("mayastorpoolcrd.yaml")
+	deleteDeployYaml("moac-rbac.yaml")
+	deleteDeployYaml("storage-class.yaml")
+	deleteDeployYaml("namespace.yaml")
+
+	// Given the yamls and the environment described in the test readme,
+	// we expect mayastor to be running on exactly 3 nodes.
```

Comment:
Copy-paste error? We are tearing down Mayastor, so we don't expect it to be running on 3 nodes.

Reply:
Done.